server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -139,7 +139,12 @@ public enum Property {
         /**
          * Indicates an index-level setting that is privately managed. Such a setting can not even be set on index creation.
          */
-        PrivateIndex
+        PrivateIndex,
+
+        /**
+         * Indicates that this index-level setting was deprecated and removed in the current major
+         */
+        DeprecatedAndRemovedInCurrentMajor
```

Comment:
We should also reject non-index settings that use this property - see `checkPropertyRequiresIndexScope`. Also I think we can make this property mutually exclusive with `Deprecated` and `DeprecatedWarning`.

Also (maybe in a followup) can we forbid `Deprecated` on `IndexScope` settings and require all index settings to use this framework instead?

Reply:
Good idea. About removing all Deprecated, I guess we'll need one more property, deprecated and removed in Next major, if we have deprecated any in 8.

Reply:
By my reckoning there's currently exactly one such index setting that's deprecated in v8 and to be removed in v9: `index.merge.policy.max_merge_at_once_explicit`.

I'll leave it up to you whether to give this setting the same treatment now or later.